consensus/safety-rules/src/safety_rules.rs

```diff
@@ -166,7 +179,7 @@ impl SafetyRules {
     }
 
     /// This verifies whether the author of one proposal is the validator signer
-    fn verify_author(&self, author: Option<Author>) -> Result<(), Error> {
+    fn verify_author(&mut self, author: Option<Author>) -> Result<(), Error> {
```

Comment:
Why is this mutable? I think we should fish out whether or not the crypto APIs should be mutable for signing rather than polluting this. If, for whatever reason, they must be, we should hide that because in general signing shouldn't be mutable. Hiding via -- rwlock -- for example.

Reply:
This is not strictly required (must have been legacy?), so it's been fixed here: https://github.com/libra/libra/pull/6762. I've rebased the PR and it looks better 😄